### PR TITLE
python.pkgs.pyqt5: 5.11.3 -> 5.12

### PR DESCRIPTION
**Pull request by dotlambda**
###### Motivation for this change
https://github.com/NixOS/nixpkgs/pull/56349#issuecomment-467031384

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by dotlambda** [comment]
Not yet ready for merge. I still have to investigate why running qutebrowser yields
```
qutebrowser needs QtWebKit or QtWebEngine, but neither could be imported!
The errors encountered were:
QtWebKit: cannot import name 'QtWebKit' from 'PyQt5' (/nix/store/5prasyaxrbcy43rdyq0zydgc1ap2l8dl-python3.7-pyqt-5.12/lib/python3.7/site-packages/PyQt5/__init__.py)
QtWebEngine: cannot import name 'QtWebEngineWidgets' from 'PyQt5' (/nix/store/5prasyaxrbcy43rdyq0zydgc1ap2l8dl-python3.7-pyqt-5.12/lib/python3.7/site-packages/PyQt5/__init__.py)
```

**Comment by dotlambda** [comment]
According to https://www.riverbankcomputing.com/news/pyqt-512, "the QtWebEngine, QtWebEngineCore and QtWebEngineWidgets modules have been moved to a separate PyQtWebEngine package." I'm currently packaging that one.

**Comment by dotlambda** [comment]
Now I'm getting
```
Fatal error: PyQt5.QtQml is required to run qutebrowser but could not be imported! Maybe it's not installed?

The error encountered was:
No module named 'PyQt5.QtQml'

Please search for the python3 version of PyQt5.QtQml in your distributions packages, or see https://github.com/qutebrowser/qutebrowser/blob/master/doc/install.asciidoc
```

File: pkgs/development/python-modules/pyqtwebengine/default.nix [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+  propagatedBuildInputs = [ pyqt5 ]
+    ++ lib.optional (!isPy3k) enum34;
+
+  configurePhase = ''
+    runHook preConfigure
+
+    mkdir -p "$out/share/sip/PyQt5"
+
+    # FIXME: Without --no-dist-info, I get
+    #     unable to create /nix/store/yv4pzx3lxk3lscq0pw3hqzs7k4x76xsm-python3-3.7.2/lib/python3.7/site-packages/PyQtWebEngine-5.12.dist-info
```
Comment (dotlambda) [inline pkgs/development/python-modules/pyqtwebengine/default.nix]: I really don't see any configure option to fix this :(

**Comment by FRidh** [comment]
Could you rebase?

**Comment by dotlambda** [comment]
Done. Note there's a new version out. However, I currently don't have time to look into this any further or work on nixpkgs at all.

**Comment by cizra** [comment]
Build fails for me:

```
$ nix-build -A qutebrowser
these derivations will be built:
  /nix/store/hcddpixnf5r6nd5acy5b1n28v71kvsha-python3.7-pyqtwebengine-5.12.1.drv
  /nix/store/aaxagq1g99520rvspgdfa5qbcx8ihhn0-qutebrowser-1.6.2.drv
building '/nix/store/hcddpixnf5r6nd5acy5b1n28v71kvsha-python3.7-pyqtwebengine-5.12.1.drv'...
unpacking sources
unpacking source archive /nix/store/bqzynfhf1d6in4cdb1bc7ni9qi74j399-PyQtWebEngine_gpl-5.12.1.tar.gz
source root is PyQtWebEngine_gpl-5.12.1
setting SOURCE_DATE_EPOCH to timestamp 1553006662 of file PyQtWebEngine_gpl-5.12.1/sip/QtWebEngineWidgets/qwebengineview.sip
patching sources
configuring
Error: Make sure you have a working sip on your PATH or use the --sip argument
to explicitly specify a working sip.
builder for '/nix/store/hcddpixnf5r6nd5acy5b1n28v71kvsha-python3.7-pyqtwebengine-5.12.1.drv' failed with exit code 1
cannot build derivation '/nix/store/aaxagq1g99520rvspgdfa5qbcx8ihhn0-qutebrowser-1.6.2.drv': 1 dependencies couldn't be built
error: build of '/nix/store/aaxagq1g99520rvspgdfa5qbcx8ihhn0-qutebrowser-1.6.2.drv' failed
```

File: pkgs/development/python-modules/pyqtwebengine/default.nix [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+  sip = pythonPackages.sip.override { sip-module = "PyQt5.sip"; };
+
+in buildPythonPackage rec {
+  pname = "pyqtwebengine";
+  version = "5.12";
+  format = "other";
+
+  src = fetchurl {
+    url = "https://www.riverbankcomputing.com/static/Downloads/PyQtWebEngine/PyQtWebEngine_gpl-${version}.tar.gz";
```
Comment (veprbl) [inline pkgs/development/python-modules/pyqtwebengine/default.nix]:
```
nix-prefetch-url "https://www.riverbankcomputing.com/static/Downloads/PyQtWebEngine/PyQtWebEngine_gpl-5.12.tar.gz"        
[0.0 MiB DL]
error: unable to download 'https://www.riverbankcomputing.com/static/Downloads/PyQtWebEngine/PyQtWebEngine_gpl-5.12.tar.gz': HTTP error 404
```
```suggestion
    url = "https://www.riverbankcomputing.com/static/Downloads/PyQtWebEngine/${version}/PyQtWebEngine_gpl-${version}.tar.gz";
```

Comment (cizra) [reply pkgs/development/python-modules/pyqtwebengine/default.nix]: That, and version 5.12 doesn't exist anymore — you need 5.12.1

File: pkgs/development/python-modules/pyqt/5.x.nix [hunk 3]
```diff
@@ -12,13 +12,13 @@ let
   sip = pythonPackages.sip.override { sip-module = "PyQt5.sip"; };
 
 in buildPythonPackage rec {
-  pname = "PyQt";
-  version = "5.11.3";
+  pname = "pyqt";
+  version = "5.12";
```
Comment (cizra) [inline pkgs/development/python-modules/pyqt/5.x.nix]:
```suggestion
  version = "5.12.1";
```

File: pkgs/development/python-modules/pyqtwebengine/default.nix (hunk 2 above)
Comment (veprbl) [reply pkgs/development/python-modules/pyqtwebengine/default.nix]: It does:
```
nix-prefetch-url https://www.riverbankcomputing.com/static/Downloads/PyQtWebEngine/5.12/PyQtWebEngine_gpl-5.12.tar.gz
[0.0 MiB DL]
path is '/nix/store/8wdwbx0vir0jkyfv4cnry1x1p1pd4625-PyQtWebEngine_gpl-5.12.tar.gz'
0j9zzgwrvh61mrzjfcdlhr08vg931ycb53ri51vynsj0grp07smn
```

Comment (cizra) [reply pkgs/development/python-modules/pyqtwebengine/default.nix]: Hah, the lack of version in path, compounded by the availability of the new version, tripped me up.

**Comment by tadeokondrak** [comment]
> Now I'm getting
> Fatal error: PyQt5.QtQml is required to run qutebrowser but could not be imported! Maybe it's not installed?

This can probably be fixed by adding more options into `pkgs/development/python-modules/pyqt/5.x.nix`. It depends on `qtbase`, but not `qtdeclarative`.

**Comment by petabyteboy** [comment]
So I tried to contiue this by updating pyqt to 5.12.2 and pyqtwebengine to 5.12.1 since old versions are not available anymore. I can not make pyqtwebengine compile: 
`mkdir: cannot create directory \u2018/nix/store/4c4ajgdnhlqk994hilagk5cgv7vw9yzg-python3-3.7.3/lib/python3.7/site-packages/PyQt5\u2019: Permission denied`
I've had some _really_ annoying bugs in qutebrowser lately, which is why I really want this update, so I can see if the mismatched pyqt version is at fault.

**Comment by FRidh** [comment]
I'm taking this further in https://github.com/NixOS/nixpkgs/pull/65836.

**Pull request closed without merging**